File: src/p1xxx/p1436.py
from typing import List, Optional


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


# Problem 1436
class Solution:
    def destCity(self, paths: List[List[str]]) -> str:
        from collections import defaultdict

        cities = set([s for _, s in paths])
        for src, _ in paths:
            cities.discard(src)

        return cities.pop()

File: src/p1xxx/p1436.rs
use core::num;

fn main() {
    println!("123456");

    use std::hint::black_box;

    println!("456789");
}

struct Solution {}

// Problem 1436
impl Solution {
    pub fn dest_city(paths: Vec<Vec<String>>) -> String {
        use std::collections::HashSet;

        let mut sources = HashSet::with_capacity(paths.len());
        let mut destinations = HashSet::with_capacity(paths.len());

        for mut pair in paths {
            destinations.insert(pair.pop().unwrap());
            sources.insert(pair.pop().unwrap());
        }

        for city in destinations {
            if sources.contains(&city) {
                continue;
            }

            return city;
        }

        unreachable!()
    }
}
